share/completions/rsync.fish

```diff
@@ -2,8 +2,24 @@ function __rsync_remote_target
     commandline -ct | string match -r '.*::?(?:.*/)?' | string unescape
 end
 
+function __rsync_parse_flags -d "Print info|help FLAGS from help output"
+    set -l helptext $argv[2..-1]
+    for line in $helptext
+        set -l tokens (string match -r "([A-Z]+)(?: {2,})(.+)" $line)
+        if [ (count $tokens) -ge 3 ]
+            echo -e "$tokens[2]\t$tokens[3]"
```

Comment:
A few nits here:

- We prefer `test` over `[ ]`
- The `echo -e` isn't necessary here, just use `echo $tokens[2]\t$tokens[3]`